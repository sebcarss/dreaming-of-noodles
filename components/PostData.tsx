export interface PostData {
    title: string;
    id: string;
    date: Date;
    tags: string[];
    published: boolean;
    excerpt: string;
    image: string;
};