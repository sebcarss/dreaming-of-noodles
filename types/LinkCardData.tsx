export interface LinkCardData {
    frontmatter: { [key: string]: any };
    content: string;
}