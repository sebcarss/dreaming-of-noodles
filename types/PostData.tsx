export interface PostData {
    title: string;
    id: string;
    date: string;
    tags: string[];
    published: boolean;
    excerpt: string;
    image: string;
};