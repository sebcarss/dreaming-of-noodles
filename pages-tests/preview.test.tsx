import Home from "../pages/preview";
import { PostData } from '../types/PostData';
import renderer from 'react-test-renderer';

// TODO: Move these to __mocks__
const title = "Dreaming of Noodles";
const allPostsData: PostData[] = [
  {
    title: "Hello World 2",
    id: "2019-01-02-hello-world-2",
    date: "2019-01-02",
    tags: ["hello", "world"],
    published: true,
    excerpt: "Hello excerpt",
    image: "https://source.unsplash.com/random/300x200",
  },
  {
    title: "Hello World 1",
    id: "2019-01-01-hello-world-1",
    date: "2019-01-01",
    tags: ["hello", "jam"],
    published: true,
    excerpt: "Hello 2 excerpt",
    image: "https://source.unsplash.com/random/300x200",
  },
];

describe("Home component", () => {
  it("should match the snapshot", () => {
    const tree = renderer.create(<Home allPostsData={allPostsData} title="Test Title" preview={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
