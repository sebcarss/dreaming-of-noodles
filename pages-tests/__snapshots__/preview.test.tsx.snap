// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should match the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <nav
      className="color-nav navbar navbar-expand-lg navbar-dark fixed-top"
    >
      <a
        className="mx-2 navbar-brand"
        href="/"
      >
        Dreaming of Noodles
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="me-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-link"
            data-rb-event-key="/japan"
            href="/japan"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Japan
          </a>
          <a
            className="nav-link"
            data-rb-event-key="/recipe-index"
            href="/recipe-index"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Recipe Index
          </a>
        </div>
      </div>
    </nav>
    <main
      className="mt-5"
    >
      <div
        className="d-flex justify-content-center"
      >
        <span
          style={
            {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": 320,
              "margin": 0,
              "opacity": 1,
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "width": 320,
            }
          }
        >
          <img
            alt="dreaming of nooodles logo"
            data-nimg="fixed"
            decoding="async"
            onError={[Function]}
            onLoad={[Function]}
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              alt="dreaming of nooodles logo"
              data-nimg="fixed"
              decoding="async"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fdreamingofnoodles.s3.eu-west-1.amazonaws.com%2Fimages%2Fdreaming-of-noodles.png&w=640&q=75"
              srcSet="/_next/image?url=https%3A%2F%2Fdreamingofnoodles.s3.eu-west-1.amazonaws.com%2Fimages%2Fdreaming-of-noodles.png&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fdreamingofnoodles.s3.eu-west-1.amazonaws.com%2Fimages%2Fdreaming-of-noodles.png&w=640&q=75 2x"
              style={
                {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </span>
      </div>
      <div
        className="mt-3 container"
      >
        <section>
          <div
            className="row"
          >
            <div
              className="col"
            >
              <div
                className="row"
              >
                <h1
                  className="text-left"
                >
                  Test Title
                </h1>
                <hr />
              </div>
              <div
                className="row"
              >
                <p>
                  Welcome to Dreaming of Noodles, my personal blog dedicated to my love of food, heavily influenced by the wonderful cuisine of Japan, and in particular, their delicious ramen. This blog will see me virtually touring the world through food, creating my own recipes, pitting cooking techniques against each other and generally keeping my stomach happy.
                </p>
                <p>
                  So come and join me on my culinary adventure!
                </p>
              </div>
            </div>
          </div>
        </section>
        <div
          className="row"
        >
          <h2
            className="text-center mt-3"
          >
            Latest Recipes
          </h2>
          <hr />
          <div
            className="col"
          >
            <div
              className="g-1 row row-cols-lg-3 row-cols-md-2 row-cols-1"
            >
              <div
                className="col"
              >
                <a
                  href="/posts/2019-01-02-hello-world-2"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <div
                    className="h-100 card bg-dark text-white"
                    style={
                      {
                        "borderRadius": 0,
                        "width": "auto",
                      }
                    }
                  >
                    <span
                      style={
                        {
                          "background": "none",
                          "border": 0,
                          "boxSizing": "border-box",
                          "display": "block",
                          "height": "initial",
                          "margin": 0,
                          "opacity": 1,
                          "overflow": "hidden",
                          "padding": 0,
                          "position": "relative",
                          "width": "initial",
                        }
                      }
                    >
                      <span
                        style={
                          {
                            "background": "none",
                            "border": 0,
                            "boxSizing": "border-box",
                            "display": "block",
                            "height": "initial",
                            "margin": 0,
                            "opacity": 1,
                            "padding": 0,
                            "paddingTop": "66.66666666666666%",
                            "width": "initial",
                          }
                        }
                      />
                      <img
                        alt="Hello World 2 - thumbnail"
                        data-nimg="responsive"
                        decoding="async"
                        onError={[Function]}
                        onLoad={[Function]}
                        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                        style={
                          {
                            "border": "none",
                            "bottom": 0,
                            "boxSizing": "border-box",
                            "display": "block",
                            "height": 0,
                            "left": 0,
                            "margin": "auto",
                            "maxHeight": "100%",
                            "maxWidth": "100%",
                            "minHeight": "100%",
                            "minWidth": "100%",
                            "objectFit": undefined,
                            "objectPosition": undefined,
                            "padding": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "width": 0,
                          }
                        }
                      />
                      <noscript>
                        <img
                          alt="Hello World 2 - thumbnail"
                          data-nimg="responsive"
                          decoding="async"
                          loading="lazy"
                          sizes="100vw"
                          src="/_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=3840&q=75"
                          srcSet="/_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=3840&q=75 3840w"
                          style={
                            {
                              "border": "none",
                              "bottom": 0,
                              "boxSizing": "border-box",
                              "display": "block",
                              "height": 0,
                              "left": 0,
                              "margin": "auto",
                              "maxHeight": "100%",
                              "maxWidth": "100%",
                              "minHeight": "100%",
                              "minWidth": "100%",
                              "objectFit": undefined,
                              "objectPosition": undefined,
                              "padding": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                              "width": 0,
                            }
                          }
                        />
                      </noscript>
                    </span>
                    <div
                      className="card-body"
                    >
                      <div
                        className="card-title h5"
                      >
                        Hello World 2
                      </div>
                      <p
                        className="text-muted card-text"
                      >
                        Hello excerpt
                      </p>
                    </div>
                  </div>
                </a>
              </div>
              <div
                className="col"
              >
                <a
                  href="/posts/2019-01-01-hello-world-1"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <div
                    className="h-100 card bg-dark text-white"
                    style={
                      {
                        "borderRadius": 0,
                        "width": "auto",
                      }
                    }
                  >
                    <span
                      style={
                        {
                          "background": "none",
                          "border": 0,
                          "boxSizing": "border-box",
                          "display": "block",
                          "height": "initial",
                          "margin": 0,
                          "opacity": 1,
                          "overflow": "hidden",
                          "padding": 0,
                          "position": "relative",
                          "width": "initial",
                        }
                      }
                    >
                      <span
                        style={
                          {
                            "background": "none",
                            "border": 0,
                            "boxSizing": "border-box",
                            "display": "block",
                            "height": "initial",
                            "margin": 0,
                            "opacity": 1,
                            "padding": 0,
                            "paddingTop": "66.66666666666666%",
                            "width": "initial",
                          }
                        }
                      />
                      <img
                        alt="Hello World 1 - thumbnail"
                        data-nimg="responsive"
                        decoding="async"
                        onError={[Function]}
                        onLoad={[Function]}
                        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                        style={
                          {
                            "border": "none",
                            "bottom": 0,
                            "boxSizing": "border-box",
                            "display": "block",
                            "height": 0,
                            "left": 0,
                            "margin": "auto",
                            "maxHeight": "100%",
                            "maxWidth": "100%",
                            "minHeight": "100%",
                            "minWidth": "100%",
                            "objectFit": undefined,
                            "objectPosition": undefined,
                            "padding": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "width": 0,
                          }
                        }
                      />
                      <noscript>
                        <img
                          alt="Hello World 1 - thumbnail"
                          data-nimg="responsive"
                          decoding="async"
                          loading="lazy"
                          sizes="100vw"
                          src="/_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=3840&q=75"
                          srcSet="/_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=3840&q=75 3840w"
                          style={
                            {
                              "border": "none",
                              "bottom": 0,
                              "boxSizing": "border-box",
                              "display": "block",
                              "height": 0,
                              "left": 0,
                              "margin": "auto",
                              "maxHeight": "100%",
                              "maxWidth": "100%",
                              "minHeight": "100%",
                              "minWidth": "100%",
                              "objectFit": undefined,
                              "objectPosition": undefined,
                              "padding": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                              "width": 0,
                            }
                          }
                        />
                      </noscript>
                    </span>
                    <div
                      className="card-body"
                    >
                      <div
                        className="card-title h5"
                      >
                        Hello World 1
                      </div>
                      <p
                        className="text-muted card-text"
                      >
                        Hello 2 excerpt
                      </p>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
  <footer
    className="mb-3"
  >
    <hr />
    <div
      className="lead text-center"
    >
      Recipes by Seb Carss
    </div>
  </footer>
</div>
`;
